### ISSUE TITLE
nspawn container "dnf upgrade" and the whole DNF system upgrade process fails, as a running systemd package cannot be updated

### ISSUE THREAD

**Issue by LaKing** [systemd/systemd#24115]
### systemd version the issue has been seen with

249

### Used distribution

Fedora 35

### Linux kernel version used

5.18.13-100.fc35.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemd-nspawn

### Expected behaviour you didn't see

It should be possible to update nspawn containers the way the host operating system is updated.
So far, this didn't work, the root cause is propably that systemd can not update itself while running.

### Unexpected behaviour you saw

Basically the distro-upgraded (34 to 35) system fails with `libsystemd-shared-249.so: cannot open shared object file: No such file or directory
`
Checking inside a shell of a running upgraded container:
```
[root@ph root]# machinectl show test2.printhub.hu
Name=test2.printhub.hu
Id=d8cd496f3a7d4843ba5dab69a0418214
Timestamp=Tue 2022-07-26 03:17:41 CEST
TimestampMonotonic=3554791845
Service=systemd-nspawn
Unit=srvctl-nspawn@test2.printhub.hu.service
Leader=43841
Class=container
RootDirectory=/srv/test2.printhub.hu/rootfs
State=running
[root@ph root]# machinectl -q --no-pager shell test2.printhub.hu
[root@test2 ~]# /usr/lib/systemd/system-generators/systemd-veritysetup-generator
/usr/lib/systemd/system-generators/systemd-veritysetup-generator: error while loading shared libraries: libsystemd-shared-249.so: cannot open shared object file: No such file or directory
[root@test2 ~]# /usr/lib/systemd/system-generators/systemd-bless-boot-generator
/usr/lib/systemd/system-generators/systemd-bless-boot-generator: error while loading shared libraries: libsystemd-shared-249.so: cannot open shared object file: No such file or directory
[root@test2 ~]# systemctl status systemd-homed.service
× systemd-homed.service - Home Area Manager
     Loaded: loaded (/usr/lib/systemd/system/systemd-homed.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Tue 2022-07-26 03:17:41 CEST; 8min ago
       Docs: man:systemd-homed.service(8)
             man:org.freedesktop.home1(5)
    Process: 29 ExecStart=/usr/lib/systemd/systemd-homed (code=exited, status=127)
   Main PID: 29 (code=exited, status=127)
        CPU: 1ms

Jul 26 03:17:41 test2.printhub.hu systemd[1]: Starting Home Area Manager...
Jul 26 03:17:41 test2.printhub.hu systemd-homed[29]: /usr/lib/systemd/systemd-homed: error while loading shared libraries: libsystemd-shared-249.so: cannot open shared object file: No such file or directory
Jul 26 03:17:41 test2.printhub.hu systemd[1]: systemd-homed.service: Main process exited, code=exited, status=127/n/a
Jul 26 03:17:41 test2.printhub.hu systemd[1]: systemd-homed.service: Failed with result 'exit-code'.
Jul 26 03:17:41 test2.printhub.hu systemd[1]: Failed to start Home Area Manager.


```

### Steps to reproduce the problem

Install an nspawn container, with fedora 34 or 35, ... then upgrade it to 35 or 36.
```
dnf upgrade --refresh
dnf install dnf-plugin-system-upgrade
dnf system-upgrade download --releasever=36
dnf system-upgrade reboot
```
Releasever is one above the current version.
After reboot the packages get updated, as they should be - except for the systemd package.
Then, when starting the container many of the services fail.

I also tried chroot-ing but since the upgrade process needs reboots, it does not seem to be an option.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd v248.3-1.fc34 running in system mode. (+PAM +AUDIT +SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified)
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: Detected virtualization systemd-nspawn.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: Detected architecture x86-64.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: 
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: Welcome to Fedora Linux 35 (Thirty Five)!
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: 
Jul 26 03:15:57 ph.d250.hu bash[41888]: [execstartpost] test2.printhub.hu
Jul 26 03:15:57 ph.d250.hu bash[41888]: [execstartpost] Calling /srv/test2.printhub.hu/ethernet.sh
Jul 26 03:15:57 ph.d250.hu bash[41897]: [ OK ] ip link set dev 30-0-5 up
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: /usr/lib/systemd/system-generators/systemd-cryptsetup-generator failed with exit status 127.
Jul 26 03:15:57 ph.d250.hu bash[41897]: [ OK ] brctl addif 10.30.0.x 30-0-5
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: /usr/lib/systemd/system-generators/systemd-veritysetup-generator failed with exit status 127.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: /usr/lib/systemd/system-generators/systemd-hibernate-resume-generator failed with exit status 127.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: /usr/lib/systemd/system-generators/systemd-bless-boot-generator failed with exit status 127.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: /usr/lib/systemd/system-generators/systemd-gpt-auto-generator failed with exit status 127.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: Queued start job for default target Graphical Interface.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Created slice system-getty.slice.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Created slice system-modprobe.slice.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Created slice system-sshd\x2dkeygen.slice.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Created slice User and Session Slice.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Dispatch Password Requests to Console Directory Watch.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Forward Password Requests to Wall Directory Watch.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Local Encrypted Volumes.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Paths.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Remote File Systems.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Slices.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Swap.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Local Verity Protected Volumes.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on Process Core Dump Socket.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on initctl Compatibility Named Pipe.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on Journal Socket (/dev/log).
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on Journal Socket.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on Network Service Netlink Socket.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on User Database Manager Socket.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Journal Service...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Remount Root and Kernel File Systems...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Apply Kernel Variables...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd-remount-fs.service: Main process exited, code=exited, status=127/n/a
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd-remount-fs.service: Failed with result 'exit-code'.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Remount Root and Kernel File Systems.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-remount-fs.service' for details.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd-sysctl.service: Main process exited, code=exited, status=127/n/a
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd-sysctl.service: Failed with result 'exit-code'.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Apply Kernel Variables.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-sysctl.service' for details.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Rebuild Hardware Database...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Create System Users...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd-hwdb-update.service: Main process exited, code=exited, status=127/n/a
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: systemd-hwdb-update.service: Failed with result 'exit-code'.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Rebuild Hardware Database.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-hwdb-update.service' for details.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Create System Users.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Create Static Device Nodes in /dev...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Journal Service.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Flush Journal to Persistent Storage...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Create Static Device Nodes in /dev.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Local File Systems (Pre).
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Local File Systems.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Restore /run/initramfs on shutdown...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Rebuild Dynamic Linker Cache...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Restore /run/initramfs on shutdown.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Rebuild Dynamic Linker Cache.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Flush Journal to Persistent Storage.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Create Volatile Files and Directories...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Create Volatile Files and Directories.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Rebuild Journal Catalog...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Userspace Out-Of-Memory (OOM) Killer...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Update UTMP about System Boot/Shutdown...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Update UTMP about System Boot/Shutdown.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Rebuild Journal Catalog.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Update is Completed...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Update is Completed.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target System Initialization.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started dnf makecache --timer.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Daily rotation of log files.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Daily Cleanup of Temporary Directories.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started daily update of the root trust anchor for DNSSEC.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Timers.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Listening on D-Bus System Message Bus Socket.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Sockets.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Basic System.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting firewalld - dynamic firewall daemon...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting System Logging Service...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target sshd-keygen.target.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting Home Area Manager...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting User Login Management...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started System Logging Service.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Home Area Manager.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-homed.service' for details.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Home Area Activation.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]:          Starting D-Bus System Message Bus...
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Userspace Out-Of-Memory (OOM) Killer.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started D-Bus System Message Bus.
Jul 26 03:15:57 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started User Login Management.
Jul 26 03:15:58 ph.d250.hu bash[41907]: Warning: ZONE_ALREADY_SET: '10.30.0.x' already bound to 'trusted'
Jul 26 03:15:58 ph.d250.hu bash[41907]: success
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started firewalld - dynamic firewall daemon.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Network (Pre).
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Configuration...
Jul 26 03:15:58 ph.d250.hu bash[41897]: [ OK ] fireall-cmd added 10.30.0.x  to the trusted interfaces
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-networkd.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Configuration...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-networkd.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Configuration...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-networkd.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Configuration...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-networkd.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Configuration...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-networkd.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Configuration.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-networkd.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Name Resolution...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Authorization Manager...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Authorization Manager.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-resolved.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Name Resolution...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-resolved.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Name Resolution...
Jul 26 03:15:58 ph.d250.hu bash[41936]: Warning: ALREADY_ENABLED: 10.30.0.x
Jul 26 03:15:58 ph.d250.hu bash[41936]: success
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-resolved.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Name Resolution...
Jul 26 03:15:58 ph.d250.hu bash[41897]: [ OK ] fireall-cmd added 10.30.0.x  to the trusted interfaces
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-resolved.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Network Name Resolution...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-resolved.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Stopped Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [FAILED] Failed to start Network Name Resolution.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: See 'systemctl status systemd-resolved.service' for details.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Network.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Network is Online.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Host and Network Name Lookups.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Dovecot IMAP/POP3 email server...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting The Apache HTTP Server...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting OpenSSH server daemon...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Permit User Sessions...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Permit User Sessions.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Console Getty.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Login Prompts.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started OpenSSH server daemon.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started The Apache HTTP Server.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Started Dovecot IMAP/POP3 email server.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Multi-User System.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Reached target Graphical Interface.
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]:          Starting Update UTMP about System Runlevel Changes...
Jul 26 03:15:58 ph.d250.hu systemd-nspawn[41846]: [  OK  ] Finished Update UTMP about System Runlevel Changes.
Jul 26 03:15:59 ph.d250.hu systemd-nspawn[41846]: 
Jul 26 03:15:59 ph.d250.hu systemd-nspawn[41846]: Fedora Linux 35 (Thirty Five)
Jul 26 03:15:59 ph.d250.hu systemd-nspawn[41846]: Kernel 5.18.13-100.fc35.x86_64 on an x86_64 (console)
Jul 26 03:15:59 ph.d250.hu systemd-nspawn[41846]:
```

**Comment by poettering** [comment]
Uh, that smells like a coruppted upgrade, i.e. a dnf run that was interrupted in the middle.

I don't see thtat nspawn had anything to do with this. I mean, it could be that nspawn was hard killed during the upgrade?

dnf is not safe in regards to abnormal aborts, it *will* break your system.

**Comment by poettering** [comment]
hmm, did you use dnf offline upgrades? I am not sure dnf can do that in containers, no idea? 

either way, this is nothing to report to npsawn really, but to the dnf people. Either they should make offline upgrades work with npsawn, or simply refuse them in a container.

There's nothing much we can do about that from our side, as dnf is just one payload for us like any other... Sorry.

**Comment by LaKing** [comment]
> Uh, that smells like a coruppted upgrade, i.e. a dnf run that was interrupted in the middle.
> I don't see thtat nspawn had anything to do with this. I mean, it could be that nspawn was hard killed during the upgrade?

Negative. I was carefully following the DNF instructions, and after the first reboot, I saw in the logs that the packages were updated, and the container rebooted itself. I did not interrupt the process, and it was really fast.

I will check offline upgrades, with these dnf commands, thanks for the hint. ..

Thank you for taking a look, greatly appreciated.
